### ISSUE TITLE
Extra newline added when using a collection expression with { get; }

### ISSUE THREAD

**Issue by SapiensAnatis** [belav/csharpier#1063]
I see some of the formatting issues with C# 12 collection expressions were fixed in #1047 but I think I have found another one.

I'm using version 0.26.4 and have found that the following code:

```csharp
public List<HttpRequestMessage> Requests { get; } =
    [
        new()
        {
            Content = new StringContent("content"),
            Headers = { { "header", "value" } },
            Method = HttpMethod.Post
        }
    ];
```

will get formatted to

```csharp
public List<HttpRequestMessage> Requests { get; } =

    [
        new()
        {
            Content = new StringContent("content"),
            Headers = { { "header", "value" } },
            Method = HttpMethod.Post
        }
    ];
```

i.e. it leaves a blank line between the property declaration and the value. This isn't consistent with the behaviour for fields or auto-properties (`=>`) which don't add a blank line so I assume it's unintentional.

**Comment by belav** [comment]
I don't remember exactly what that does, but the [tests](https://github.com/belav/csharpier/blob/main/Src/CSharpier.Tests/FormattingTests/TestFiles/cs/CollectionExpressions.test) should hopefully catch anything that you break. I was thinking it had to do with the ExpressionBody property in there, but I don't believe that contains an EqualsValueClause.

I also use this [script](https://github.com/belav/csharpier/blob/main/Shell/ReviewBranch.psm1) to test changes against https://github.com/belav/csharpier-repos. The script probably requires changes for someone else to run.

TLDR, I put things in place to make sure that I also don't end up breaking stuff, so feel free to give it a shot!

**Comment by SapiensAnatis** [comment]
.assign
